src/main/java/at/favre/lib/crypto/HKDF.java

```diff
@@ -148,6 +163,21 @@ public static HKDF from(HkdfMacFactory macFactory) {
      * @see <a href="https://tools.ietf.org/html/rfc5869#section-2.3">RFC 5869 Section 2.3</a>
      */
     public byte[] expand(byte[] pseudoRandomKey, byte[] info, int outLengthBytes) {
+        return expand(macFactory.createSecretKey(pseudoRandomKey), info, outLengthBytes);
+    }
+
+    /**
+     * Use this if you require {@link SecretKey} types by your security framework. See also
+     * {@link HkdfMacFactory#createSecretKey(byte[])}.
+     * <p>
+     * See {@link #expand(byte[], byte[], int)} for description.
+     *
+     * @param pseudoRandomKey a pseudo random key of at least hmac hash length in bytes (usually, the output from the extract step)
```

Comment:
But if you have `byte[]` returned from extract step, you would need to create a `SecretKey` out of it, right?

Reply:
Correct - however this does not seem intuitive I believe it makes sense. The `expand` returns a piece of entropy which always will be a byte array known to the JVM when using this lib (the output of a HMAC basically).

The `extract` API using the SecretKey type is a special use case (I just wanted to include it to make the API complete), most times the `extract` will be used with the byte[] API even when using a HSM.